enhancements/insights/insights-config-api.md

```diff
@@ -0,0 +1,315 @@
+	DisabledGatherers []string `json:"disabledGatherers"`
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "NoPolicy"
+	// IP addresses and cluster domain name is obfuscated
+	IPsAndClusterDomainPolicy DataPolicy = "IPsAndClusterDomainPolicy"
+)
+
+// DataPolicy is a valid value for GatheringConfig.DataPolicy
+type DataPolicy string
+```
+
+The proposal for the new `insights.operator.openshift.io` resource is:
+
+```go
+type InsightsSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus *GatheringStatus `json:"gatheringStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus *ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatherTime is the last time when Insights gathering finished.
+	LastGatherTime metav1.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status *GatherStatus `json:"status,omitempty"`
+	// List of active gatherers (and their statuses) in the last gathering.
+	GathererStatuses []GathererStatus `json:"gathererStatuses,omitempty"`
+}
+
+type ReportStatus struct {
+	// Number of active Insights healthchecks with low severity
+	LowCount int `json:"low"`
+	// Number of active Insights healthchecks with moderate severity
+	ModerateCount int `json:"moderate"`
+	// Number of active Insights healthchecks with important severity
+	ImportantCount int `json:"important"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total"`
+}
+
+type GathererStatus struct {
+	// Name is the name of the gatherer.
+	Name string `json:"name"`
+	// GathererConditions provide details on the status of each gatherer.
+	GathererConditions []GathererCondition `json:"conditions"`
+	// DurationMillisecond represents the time spent gathering.
+	DurationMillisecond int64 `json:"durationMillisecond"`
+	// RecordsCount represents the number of records produced by the gatherer.
+	RecordsCount int `json:"recordsCount"`
+}
+
+type GathererCondition struct {
+	// Type of the gatherer condition
+	Type GathererConditionType `json:"type"`
+	// Status is last known status of the particular gatherer condition
+	Status corev1.ConditionStatus `json:"status"`
+	// Messages is an optional attribute that provides error and warning messages
+	// from the gatherer
+	Messages []string `json:"messages,omitempty"`
```

Comment:
Added.

Reply:
I didn't notice that earlier, but since each gatherer has its own status `GathererStatus`, and within that you have an array of conditions `GathererConditions` you could probably change this Messages array to be just `Message string` I don't expect that much of data in it, based on my previous experience.

Reply:
This is the same as the https://github.com/openshift/enhancements/pull/1037#discussion_r844961191. I think it's more convenient as a slice for our purposes. We can have more various messages as in https://github.com/openshift/insights-operator/blob/master/docs/insights-archive-sample/insights-operator/gathers.json#L7